api/src/main/java/org/openmrs/api/impl/CohortServiceImpl.java

```diff
@@ -200,27 +199,25 @@ public Cohort removeMemberShipFromCohort(Cohort cohort, CohortMembership cohortM
 	public void patientVoided(Patient patient) throws APIException {
 		List<Cohort> cohorts = Context.getCohortService().getCohortsContainingPatient(patient);
 		for (Cohort cohort : cohorts) {
-			List<CohortMembership> membersToVoid = cohort.getMembers().stream()
-					.filter(m -> !m.getVoided() && patient.getVoided() && m.getPatient().getPatientId().equals(patient.getPatientId()))
-					.collect(Collectors.toList());
-			for (CohortMembership member : membersToVoid) {
-				member.setVoided(patient.getVoided());
-				member.setDateVoided(patient.getDateVoided());
-				member.setVoidedBy(patient.getVoidedBy());
-				member.setVoidReason(patient.getVoidReason());
+			if (patient.getVoided()) {
+				CohortMembership membershipToVoid = cohort.getActiveMembership(patient);
```

Comment:
You should call getMemberships() and void those that are not yet voided instead of only active ones, didn't you add a `getMemberships(includeVoided)` method that you can call?